src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -2643,6 +2646,123 @@ class TypeSpecTest {
     }.hasMessageThat().isEqualTo("ANNOTATION Taco.eat requires modifiers [PUBLIC, ABSTRACT]")
   }
 
+  @Test fun basicDelegateTest() {
+    val type = TypeSpec.classBuilder("Guac")
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("somethingElse", String::class)
+            .build())
+        .addSuperinterface(
+            ParameterizedTypeName.get(Consumer::class, String::class),
+            CodeBlock.of("({ println(it) })"))
+        .build()
+
+    val expect = """
+        |package com.squareup.tacos
+        |
+        |import java.util.function.Consumer
+        |import kotlin.String
+        |
+        |class Guac(somethingElse: String) : Consumer<String> by ({ println(it) })
+        |""".trimMargin()
+
+    assertThat(toString(type)).isEqualTo(expect)
+  }
+
+  @Test fun testMultipleDelegates() {
+    val type = TypeSpec.classBuilder("Guac")
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter("somethingElse", String::class)
+            .build())
+        .addSuperinterface(ClassName.bestGuess("RandomIface"), CodeBlock.of("Factory.byStrategy(FacadeImpl)"))
+        .addSuperinterface(Runnable::class, CodeBlock.of("Runnable({ Logger.debug(\"Hello world\") })"))
```

Comment:
I don’t think this is a good test because it shows a type `Logger` being embedded directly rather than with the proper syntax `%T`. We expect our users to read our test cases and use them as examples, and this is one where we’re taking a shortcut we don’t want to take.

The big consequence is that imports don’t work.